caffe2/CMakeLists.txt

```diff
@@ -614,7 +614,7 @@ if(NOT INTERN_BUILD_MOBILE OR NOT BUILD_CAFFE2_MOBILE)
       set(DELAY_LOAD_FLAGS "")
     endif()
     target_link_libraries(caffe2_nvrtc ${CUDA_NVRTC} ${CUDA_CUDA_LIB} ${CUDA_NVRTC_LIB} ${DELAY_LOAD_FLAGS})
-    target_include_directories(caffe2_nvrtc PRIVATE ${CUDA_INCLUDE_DIRS})
+    target_include_directories(caffe2_nvrtc PRIVATE ${CUDA_INCLUDE_DIRS} ${CMAKE_BINARY_DIR})
```

Comment:
Can you please explain why this change is necessary?

Reply:
Because this target compiles `ATenNVRTC.cpp`, which includes `ATenNVRTC.h` which includes `ATen/cuda/ATenCUDAGeneral.h` which includes... `c10/macros/Export.h`, generated in `${CMAKE_BINARY_DIR}`.

Actually, I believe that `caffe2_nvrtc` target in ROCm branch also requires this fix.

Reply:
I'm wondering if `c10` shouldn't be added to dependencies of `caffe2_nvrtc` through `target_link_libraries()` instead of this `target_include_directories()`, it would make more sense, specially in ROCm branch where several `c10` headers are included.